File: Sesion 12/Actividades/Actividad 2 S12/ACTIVIDAD2 S12.cpp
#include <iostream>
#include <list>
#include <algorithm>
#include <iterator>
#include <cctype>

using namespace std;

void printList(const list<string>& words) {
    for (const auto& word : words) {
        cout << word << " ";
    }
    cout << endl;
}

int main() {
    list<string> words;
    string input;

    cout << "Ingrese palabras (ingrese 'fin' para terminar):" << endl;
    while (true) {
        cin >> input;
        if (input == "fin") {
            break;
        }
        words.push_back(input);
    }

    cout << "Lista original: ";
    printList(words);

    words.sort();
    cout << "Lista ordenada alfabeticamente: ";
    printList(words);

    cout << "Ingrese la palabra a contar: ";
    string word_to_count;
    cin >> word_to_count;

    int count = std::count(words.begin(), words.end(), word_to_count);
    cout << "La palabra '" << word_to_count << "' aparece " << count << " veces en la lista." << endl;

    transform(words.begin(), words.end(), words.begin(), [](string& word) {
        transform(word.begin(), word.end(), word.begin(), ::toupper);
        return word;
    });

    cout << "Lista en mayúsculas: ";
    printList(words);

    return 0;
}

File: Sesion 12/Actividades/Actividad 2 S12/ACTIVIDAD2 S12.py
def print_list(words):
    print(" ".join(words))

def main():
    words = []
    print("Ingrese palabras (ingrese 'fin' para terminar):")
    while True:
        word = input()
        if word.lower() == "fin":
            break
        words.append(word)

    print("Lista original:")
    print_list(words)

    words.sort()
    print("Lista ordenada alfabeticamente:")
    print_list(words)

    word_to_count = input("Ingrese la palabra a contar: ")
    count = words.count(word_to_count)
    print(f"La palabra '{word_to_count}' aparece {count} veces en la lista.")

    words = [word.upper() for word in words]
    print("Lista en mayúsculas:")
    print_list(words)

if __name__ == "__main__":
    main()
